ui/src/lib/reactFlow/useNodeSchemaGenerateHook.ts

```diff
@@ -0,0 +1,110 @@
+import { RJSFSchema } from "@rjsf/utils";
+import { useTranslation } from "react-i18next"; // Assuming you're using react-i18next
+
+const useNodeSchemaGenerate = (
+  nodeType: string,
+  officialName: string,
+): RJSFSchema => {
+  const { t } = useTranslation();
```

Comment:
This isn't our standard. If you are using AI, please be thorough in reviewing and updating the code